sw/device/tests/sim_dv/sram_ctrl_main_scrambled_access_test.c

```diff
@@ -290,17 +300,17 @@ bool test_main(void) {
 
   if (reentry_after_system_reset()) {
     // Second boot, check the values.
-    CHECK(sram_ctrl_testutils_read_check_neq(
-        TOP_EARLGREY_SRAM_CTRL_RET_AON_RAM_BASE_ADDR, &kRamTestPattern1));
-    CHECK(sram_ctrl_testutils_read_check_neq(
-        TOP_EARLGREY_SRAM_CTRL_RET_AON_RAM_BASE_ADDR, &kRamTestPattern2));
+    CHECK_ARRAYS_NE((uint32_t *)kRetSramAddr, kRamTestPattern1,
+                    ARRAYSIZE(kRamTestPattern1));
+
+    CHECK_ARRAYS_NE((uint32_t *)kRetSramAddr, kRamTestPattern2,
+                    ARRAYSIZE(kRamTestPattern2));
 
     // Statistically there is always a chance that after changing the scrambling
     // key the ECC bits are correct and no IRQ is triggered. So we tolerate a
     // minimum of false positives.
     uint32_t ecc_errors = mmio_region_read32(
-        mmio_region_from_addr(TOP_EARLGREY_SRAM_CTRL_RET_AON_RAM_BASE_ADDR),
-        ECC_ERROR_COUNT_OFFSET);
+        mmio_region_from_addr(kRetSramAddr), ECC_ERROR_COUNT_OFFSET);
     uint32_t false_positives = SRAM_CTRL_TESTUTILS_DATA_NUM_WORDS - ecc_errors;
```

Comment:
```suggestion
    uint32_t false_positives = ARRAYSIZE(something) - ecc_errors;
```

should probably add an assertion that both arrays are the same size.

Reply:
done